salt/_modules/vault.py

```diff
@@ -0,0 +1,858 @@
+                    )
+
+            # Build return object
+            ret['changes']['old'] =[ob.type for ob in remote_methods]
+            ret['changes']['new'] = new_auth_methods or "No changes"
+
+        except Exception:
+            raise
+
+        log.info('Finished processing and configuring auth methods...')
+
+    def cleanup_auth_methods(self, client, remote_methods, local_methods, ret):
+        """Disabling any auth methods not present in the local config
+
+        Arguments:
+            client {hvac} -- hvac client
+            remote_methods {list} -- auth methods from remote vault server
+            local_methods {list} -- auth methods from local config
+            ret {dict} -- salt state result
+        """
+        log.info('Cleaning up auth methods...')
+        has_change = False
+
+        try:
+            for auth_method in remote_methods:
+                if auth_method not in local_methods:
+                    has_change = True
+                    log.debug(
+                        'Auth method "%s" does not exist in configuration. Disabling...', auth_method.type)
+                    client.sys.disable_auth_method(
+                        path=auth_method.path
+                    )
+                    log.debug('Auth method "%s" is disabled.', auth_method.type)
+
+            if has_change:
+                ret['changes']['new'] = [ob.type for ob in local_methods]
+        except Exception:
+            raise
+
+        log.info('Finished cleaning up auth methods.')
+
+
+class VaultSecretsManager():
+    """
+    Module for handling Vault Secrets Engines
+    """
+
+    def __init__(self):
+        """Initialize Vault Secrets Manager
+        """
+        log.info("Initializing Vault Secret Manager...")
+
+    def get_remote_secrets_engines(self, client, ret):
+        """Retrieve secrets engines from remote vault server
+
+        Arguments:
+            client {hvac} -- hvac client
+            ret {dict} -- salt state result
+
+        Returns:
+            list -- secrets engines
+        """
+        log.info('Retrieving secrets engines from Vault')
+        remote_secret_engines = []
+        try:
+            secrets_engines_resp = client.sys.list_mounted_secrets_engines()
+            for engine in secrets_engines_resp['data']:
+                remote_secret_engines.append(
+                    VaultSecretEngine(
+                        type=secrets_engines_resp[engine]['type'],
+                        path=(secrets_engines_resp[engine]["path"]
+                              if 'path' in secrets_engines_resp[engine] else engine),
+                        description=secrets_engines_resp[engine]["description"],
+                        config=OrderedDict(
+                            sorted(secrets_engines_resp[engine]["config"].items()))
+                    )
+                )
+            remote_secret_engines.sort(key=lambda x: x.type)
+        except Exception:
```

Comment:
Since we are not actually handling any exception, just re-raising, this try/except block is not needed. Can remove it and let things raise exceptions normally.